pubsub/adapters/googlecloud.py

```diff
@@ -26,15 +30,31 @@ def ack(self, message):
         message.ack()
 
     def publish(self, topic_name, message):
-        topic_path = self.publisher.topic_path(self.project_id, topic_name)
-        self.get_topic(topic_path)
-        self.publisher.publish(topic_path, message)
+        if self.pubsub_rest_proxy:
+            self.bulk_publish(topic_name, [message])
+        else:
+            topic_path = self.publisher.topic_path(self.project_id, topic_name)
+            self.get_topic(topic_path)
+            self.publisher.publish(topic_path, message)
+
+    def bulk_publish(self, topic_name, messages):
+        if self.pubsub_rest_proxy:
+            # We could use json=, but the message is already dumped and encoded
+            response = self.session.post(
+                '{}/messages/{}'.format(self.pubsub_rest_proxy, topic_name),
+                data=messages, headers={'Content-Type': 'application/json'})
```

Comment:
```suggestion
                json=messages)
```
If it is just using the standard python requests module you can do this.
For rest-proxy I also had to add some extra formatting because pubsub-rest-proxy tries to get the key messages:
`r = requests.post(f"http://pubsub-rest-proxy/messages/{topic}", 
  json={'messages': [data]})`

Reply:
Hmm OK, I didn't actually change any of this, there is a comment on line 42 about using `json=` which I didn't really understand - I'll make the change

Reply:
As discussed this happens further up the chain in the `Protocol`